lib/util.js

```diff
@@ -403,8 +403,9 @@ function formatValue(ctx, value, recurseTimes) {
   // Some type of object without properties can be shortcutted.
   if (keys.length === 0) {
     if (typeof value === 'function') {
-      return ctx.stylize(`[Function${value.name ? `: ${value.name}` : ''}]`,
-                         'special');
+      const cName = getConstructorOf(value).name;
```

Comment:
Sorry to comment again :smile: I’d just move the `constructor` variable up before the `keys.length` block and do the same `constructor && constructor.name …` check that is used elsewhere in this file. I doubt there are many people who set their functions’ prototypes to `null`, but I’d prefer handling these consistently…